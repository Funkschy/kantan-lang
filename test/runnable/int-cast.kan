import "io";

def main() {
    let i: i64 = 42;
    let i = i as i32;
    let i = i as i16;

    io.printf("%d %d\n", sizeof i16, i);
    io.printf("%d %d\n", sizeof i32, i as i32);
    io.printf("%d %d\n", sizeof i64, i as i64);

    i = 'A';
    io.printf("%d %d\n", sizeof i16, i);
}
