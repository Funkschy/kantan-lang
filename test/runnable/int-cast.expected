2 42
4 42
8 42
